sklearn/metrics/tests/test_common.py

```diff
@@ -240,7 +241,10 @@ def precision_recall_curve_padded_thresholds(*args, **kwargs):
     "label_ranking_average_precision_score":
     label_ranking_average_precision_score,
     "ndcg_score": ndcg_score,
-    "dcg_score": dcg_score
+    "dcg_score": dcg_score,
+
+    "top_k_accuracy_score": top_k_accuracy_score,
+    "top_k_accuracy_score_binary": partial(top_k_accuracy_score, k=1)
```

Comment:
Could someone please comment why this is necessary. I couldn't find a reference to it in the above discussion. Wouldn't it be equal to `accuracy_score` except that it takes a probability instead of the class label as input? What's the use case for it?

Also it doesn't follow the convention "*_score"

Reply:
Thanks for raising this point Roman! The reason was because `test_binary_sample_weight_invariance` naturally fails for any `k>1`. My fix was clumsy as it doubled some tests and needed to add `top_k_accuracy_score` to `METRIC_UNDEFINED_BINARY`.

Instead, I removed `top_k_accuracy_score_binary` and added the following to `test_binary_sample_weight_invariance`:

```python
# top_k_accuracy_score always lead to a perfect score for k > 1 in the
# binary case
metric = partial(metric, k=1) if name == "top_k_accuracy_score" else metric
```